File: InterviewProblems/IsPalindrome/IsPalindrome.py
# Problem:
# Check if a string is a palindrome or not.
# Ex) is_palindrome(level) -> True | is_palindrome(levels) -> False

def is_palindrome(string):
    i = 0
    j = len(string) - 1
    while i <= j:
        if string[i] != string[j]:
            return False
        i += 1
        j -= 1
    return True


def main():
    test1 = 'level'
    test2 = 'levels'
    print('level:', is_palindrome(test1))
    print('levels:', is_palindrome(test2))


main()

File: InterviewProblems/IsPalindrome/IsPalindrome.java
package InterviewProblems.IsPalindrome;

/*
Problem:
Check if a string is a palindrome or not.
Ex) is_palindrome(level) -> True | is_palindrome(levels) -> False
*/

public class IsPalindrome {

    public static boolean isPalindrome(String str) {
        int i = 0;
        int j = str.length() - 1;
        while (i <= j) {
            if (str.charAt(i) != str.charAt(j)) return false;
            ++i;
            --j;
        }
        return true;
    }

    // Test Client
    public static void main(String[] args) {
        String test1 = "level";
        String test2 = "levels";
        System.out.println("level: " + IsPalindrome.isPalindrome(test1));
        System.out.println("levels: " + IsPalindrome.isPalindrome(test2));
    }
}